# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
# Copyright (c) 2021-2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0

import launch
from launch_ros.actions import ComposableNodeContainer
from launch_ros.descriptions import ComposableNode


def generate_launch_description():
    crop_node = ComposableNode(
        package='isaac_ros_image_proc',
        plugin='nvidia::isaac_ros::image_proc::CropNode',
        name='crop',
        parameters=[{
            'input_width': 1920,
            'input_height': 1200,
            'crop_width': 1200,
            'crop_height': 720,
            'crop_mode': 'CENTER'
        }],
        remappings=[
            ('image', 'hawk_0_left_rgb_image'),
            ('camera_info', 'hawk_0_left_rgb_camera_info'),
        ])

    crop_container = ComposableNodeContainer(
        package='rclcpp_components',
        name='crop_container',
        namespace='',
        executable='component_container_mt',
        composable_node_descriptions=[
            crop_node,
        ],
        output='screen'
    )

    return launch.LaunchDescription([crop_container])
